packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -490,17 +490,21 @@ export class ContextualMenuBase extends React.Component<IContextualMenuProps, IC
     );
   };
 
-  private _renderMenuItem = (item: IContextualMenuItemRenderProps): React.ReactNode => {
+  // While we have access to the underlying menu item, we should not mutate or expand it into a new object as it will
+  // cause the menu items to always rerender even if the same props object is passed in.
```

Comment:
```suggestion
  /**
   * !!!IMPORTANT!!! Avoid mutating `item: IContextualMenuItem` argument. It will
   * cause the menu items to always re-render because the component update is based on shallow comparison.
   */
```